### PR TITLE
Add Data Loss Prevention (DLP) architecture

### PR DESCRIPTION
**Pull request by claick-oliveira**
This pull request introduces a new architecture for protecting sensitive data using Google Cloud's Data Loss Prevention (DLP) service.

**Key features:**

- **Cloud Run deployment:** The architecture leverages Cloud Run to deploy two services:
  - **Generate Service:** Generates logs containing sensitive data.
  - **Redact Service:** Redacts sensitive data from logs using DLP.
- **Log Router and Pub/Sub integration:** Logs from Generate Service are routed to Pub/Sub, which triggers Redact Service to mask sensitive information.
- **Terraform for infrastructure management:** Terraform is used to automate the deployment and management of the infrastructure.
- **Comprehensive documentation:** The README.md file provides detailed instructions on deploying, testing, and cleaning up the architecture.

**Benefits:**

- **Automated data protection:** The architecture automatically redacts sensitive data from logs, reducing the risk of data breaches.
- **Scalability and reliability:** Cloud Run and Pub/Sub ensure scalability and reliability for handling large volumes of logs.
- **Simplified deployment and management:** Terraform simplifies the deployment and management of the infrastructure.

This pull request includes:

- Updated README.md with detailed instructions and explanations.
- Code for Generate Service and Redact Service.
- Terraform configuration files for infrastructure deployment.

### CODE REVIEW

File: data-loss-prevention/code/generator/.vscode/extensions.json
[the remark this answers is not in the document]
Comment (claick-oliveira) [reply data-loss-prevention/code/generator/.vscode/extensions.json:1]: Deleted

File: data-loss-prevention/code/generator/.vscode/tasks.json
[the remark this answers is not in the document]
Comment (claick-oliveira) [reply data-loss-prevention/code/generator/.vscode/tasks.json:1]: Deleted

File: data-loss-prevention/code/redact/.vscode/extensions.json
[the remark this answers is not in the document]
Comment (claick-oliveira) [reply data-loss-prevention/code/redact/.vscode/extensions.json:1]: Deleted

File: data-loss-prevention/code/redact/.vscode/tasks.json
[the remark this answers is not in the document]
Comment (claick-oliveira) [reply data-loss-prevention/code/redact/.vscode/tasks.json:1]: Deleted

**Review by fellipeamedeiros — APPROVED** [review]
LGTM

**Pull request merged**